spring-kafka/src/main/java/org/springframework/kafka/support/micrometer/KafkaListenerObservation.java

```diff
@@ -64,18 +84,151 @@ public enum ListenerLowCardinalityTags implements KeyName {
 		LISTENER_ID {
 
 			@Override
+			@NonNull
 			public String asString() {
 				return "spring.kafka.listener.id";
 			}
 
-		}
+		},
+
+		/**
+		 * Messaging system.
+		 */
+		MESSAGING_SYSTEM {
+
+			@Override
+			@NonNull
+			public String asString() {
+				return "messaging.system";
+			}
+
+		},
+
+		/**
+		 * Messaging operation.
+		 */
+		MESSAGING_OPERATION {
+
+			@Override
+			@NonNull
+			public String asString() {
+				return "messaging.operation";
+			}
+
+		},
+
+		/**
+		 * Messaging source name.
+		 */
+		MESSAGING_SOURCE_NAME {
+
+			@Override
+			@NonNull
+			public String asString() {
+				return "messaging.source.name";
+			}
+
+		},
+
+		/**
+		 * Messaging source kind.
+		 */
+		MESSAGING_SOURCE_KIND {
+
+			@Override
+			@NonNull
+			public String asString() {
+				return "messaging.source.kind";
+			}
+
+		},
+
+		/**
+		 * Messaging the consumer group.
+		 */
+		MESSAGING_CONSUMER_GROUP {
```

Comment:
I believe all of these new `enum` values must be marked with `@since 3.2`.
Because they are addition to the existing class.